src/content/docs/en/recipes/making-toolbar-apps.mdx

```diff
@@ -0,0 +1,322 @@
+
+    const motivationalMessages = [
+      "You're doing great!",
+      "Keep up the good work!",
+      "You're awesome!",
+      "You're a star!",
+    ];
+
+    export default defineToolbarApp({
+        init(canvas) {
+          const h1 = document.createElement('h1');
+          h1.textContent = motivationalMessages[Math.floor(Math.random() * motivationalMessages.length)];
+
+          canvas.append(h1);
+        },
+    });
+    ```
+
+2. Start the dev server if it is not already running and toggle the app on in the dev toolbar. If your app is working successfully, you will see a motivational message displayed in the top-left corner of the screen. (And, it's true!) 
+
+    However, this message will not change when the app is toggled on and off, as the `init()` function is only called once when the app is loaded.
+
+3. To add client-side interactivity to your app, add the `app` argument and use `onAppToggled()` to select a new random message each time your toolbar app is toggled on:
+
+    ```ts title="app.ts" ins=", app" ins={17-21}
+    import { defineToolbarApp } from "astro/toolbar";
+
+    const motivationalMessages = [
+      "You're doing great!",
+      "Keep up the good work!",
+      "You're awesome!",
+      "You're a star!",
+    ];
+
+    export default defineToolbarApp({
+        init(canvas, app) {
+          const h1 = document.createElement('h1');
+          h1.textContent = motivationalMessages[Math.floor(Math.random() * motivationalMessages.length)];
+          
+          canvas.append(h1);
+
+          // Display a random message when the app is toggled
+          app.onToggled(({ state }) => {
+            const newMessage = motivationalMessages[Math.floor(Math.random() * motivationalMessages.length)];
+            h1.textContent = newMessage;
+          });
+        },
+    });
+    ```
+
+4. In your browser preview, toggle your app on and off several times. With this change, a new random message will be selected every time you toggle the app on, providing you with an infinite source of motivation!
+
+</Steps>
+
+<ReadMore>See the [Astro Dev Toolbar API documentation](/en/reference/dev-toolbar-app-reference/) for more about building dev toolbar apps.</ReadMore>
+
+## Building apps with a UI framework
+
+UI frameworks like React, Vue, or Svelte can also be used to create dev toolbar apps. These frameworks provide a more declarative way to create UIs and can make your code more maintainable and easier to read.
+
+The same motivational dev toolbar app built into your existing Astro project earlier on this page with JavaScript can be built instead using a UI framework (e.g. Preact). Depending on your chosen framework, you may or may not require a build step.
+
+:::note 
+However your choose to build your dev toolbar app, using JavaScript or a UI framework, you will still need to [create the integration](#creating-the-astro-integration) that adds your app to the dev toolbar.
```

Comment:
«However your choice... »?
«However you choose... »?

Reply:
Great catch!